src/chains/filecoin/filecoin/src/blockchain.ts

```diff
@@ -188,11 +191,39 @@ export default class Blockchain extends Emittery.Typed<
   }
 
   private async getIPFSObjectSize(cid: string): Promise<number> {
-    let stat = await this.ipfsServer.node.object.stat(cid, {
+    let stat = await this.ipfsServer.node.object.stat(cid as any, {
       timeout: 500 // Enforce a timeout; otherwise will hang if CID not found
     });
 
-    return stat.DataSize;
+    return stat.CumulativeSize;
+  }
+
+  private async downloadFile(cid: string, ref: FileRef): Promise<void> {
+    const size = await this.getIPFSObjectSize(cid);
+    const content = new Uint8Array(size);
+    const chunks = this.ipfsServer.node.files.read(new IPFS_CID(cid), {
+      timeout: 500 // Enforce a timeout; otherwise will hang if CID not found
+    });
+
+    let index = 0;
+    for await (const chunk of chunks) {
+      content.set(chunk, index);
```

Comment:
I'm a little worried that a large file could clobber the heap. Should we update this to pipe each chunk directly to a file?

Reply:
Sure!